OpenSim/Simulation/Model/ForceApplier.h

```diff
@@ -0,0 +1,75 @@
+#ifndef OPENSIM_FORCE_APPLIER_H_
+#define OPENSIM_FORCE_APPLIER_H_
+
+/* -------------------------------------------------------------------------- *
+ *                         OpenSim: ForceApplier.h                            *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2024 Stanford University and the Authors                *
+ * Author(s): Adam Kewley                                                     *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include <OpenSim/Simulation/Model/ForceConsumer.h>
+
+#include <OpenSim/Simulation/osimSimulationDLL.h>
+
+#include <SimTKcommon/internal/MassProperties.h>  // for `SimTK::SpatialVec`
+#include <SimTKcommon/internal/Vector_.h>         // for `SimTK::Vector_`
+
+namespace SimTK { class SimbodyMatterSubsystem; }
+
+namespace OpenSim
+{
+
+/**
+ * A `ForceApplier` is a concrete `ForceConsumer` implementation that:
+ *
+ * - Converts any point forces into body forces
+ * - Applies body forces to a `SimTK::Vector_<SimTK::SpatialVec>`
+ * - Applies generalized (mobility) forces to a `SimTK::Vector`
+ *
+ * The `ForceApplier` is primarily used as an internal class for adapting
+ * the `OpenSim::ForceConsumer`'s API contract ("undefined virtual
+ * consumption") to the `OpenSim::Force`'s API contract ("applies forces
+ * to a multibody system").
+ */
+class OSIMSIMULATION_API ForceApplier final : public ForceConsumer {
+public:
+    explicit ForceApplier(
```

Comment:
A generic C++ question:

If you have a ctor taking references like this, do you then expect it to store pointers to those references?
For the SimbodyMatterSubsystem you see this everywhere and come to expect it, but for the Vector arguments I would not expect it by default.

Reply:
Changed to pointers by [9c8d189](https://github.com/opensim-org/opensim-core/pull/3891/commits/9c8d189d286f8f0e0fed0929f1afa0bd616a6c1d)